lib/utils/schema.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const _ = require('lodash')
+const Joi = require('joi')
+const Hoek = require('hoek')
+
+module.exports = function() {
+    const toJoiPostValidatation = function(schema) {
```

Comment:
doesn't validate things like 'id', 'type' and 'relationships' currently
http://jsonapi.org/format/#document-resource-objects